spec/package_json_spec.rb

```diff
@@ -55,15 +55,55 @@
         end
       end
 
-      it "supports yarn classic" do
+      it "supports yarn classic with version 1.2.3" do
         with_package_json_file({ "version" => "1.0.0", "packageManager" => "yarn@1.2.3" }) do
           package_json = described_class.read
 
           expect(package_json.manager).to be_a PackageJson::Managers::YarnClassicLike
         end
       end
 
-      it "supports yarn berry" do
+      it "supports yarn classic with just a major version" do
+        with_package_json_file({ "version" => "1.0.0", "packageManager" => "yarn@1" }) do
+          package_json = described_class.read
+
+          expect(package_json.manager).to be_a PackageJson::Managers::YarnClassicLike
+        end
+      end
+
+      it "supports yarn classic with a caret constraint" do
+        with_package_json_file({ "version" => "1.0.0", "packageManager" => "yarn@^1.2" }) do
+          package_json = described_class.read
+
+          expect(package_json.manager).to be_a PackageJson::Managers::YarnClassicLike
+        end
+      end
+
+      it "supports yarn classic with tilde constraint" do
+        with_package_json_file({ "version" => "1.0.0", "packageManager" => "yarn@~1.2" }) do
+          package_json = described_class.read
+
+          expect(package_json.manager).to be_a PackageJson::Managers::YarnClassicLike
+        end
+      end
+
+      it "supports yarn classic with exact version" do
+        with_package_json_file({ "version" => "1.0.0", "packageManager" => "yarn@=1.2" }) do
+          package_json = described_class.read
+
+          expect(package_json.manager).to be_a PackageJson::Managers::YarnClassicLike
+        end
+      end
+
+      it "doesn't return yarn classic if the major version is 11" do
```

Comment:
I prefer to avoid using `'` in test names to make it easier to match on the command line, and this is consistent with the wording of other tests:

```suggestion
      it "does not return yarn classic if the major version is 11" do
```